docs/site/Calling-other-APIs-and-Web-Services.md

```diff
@@ -9,9 +9,145 @@ permalink: /doc/en/lb4/Calling-other-APIs-and-web-services.html
 Your API implementation often needs to interact with REST APIs, SOAP Web
 Services, gRPC microservices, or other forms of APIs.
 
-To facilitate calling other APIs or web services, we introduce
-`@loopback/service-proxy` module to provide a common set of interfaces for
-interacting with backend services.
+To facilitate calling other APIs or web services, you can create a Service class
+to provide a common set of interfaces for interacting with backend services.
+
+There are 3 major steps:
+
+1. **Add a datasource**: specify the service you're trying to connect
+2. **Add a service**: define how the operations/methods in the external APIs
+   will be mapped to the service methods.
+3. **Add a controller**: expose the REST APIs that will call the service methods
+
+## Calling SOAP web services
+
+### Add a datasource
+
+Add a datasource with the `SOAP webservices` connector using the
+[Datasource generator](DataSource-generator.md). You also need to know the URL
+of the SOAP web service endpoint and its corresponding WSDL file.
+
+```sh
+$ lb4 datasource
+? Datasource name: ds
+? Select the connector for ds: SOAP webservices (supported by StrongLoop)
+? URL to the SOAP web service endpoint: http://calculator-webservice.mybluemix.net/calculator
+? HTTP URL or local file system path to the WSDL file: http://calculator-webservice.mybluemix.net/calculator?wsdl
+? Expose operations as REST APIs: Yes
+? Maps WSDL binding operations to Node.js methods:
+```
+
+For the last option `Maps WSDL binding operations to Node.js methods`, specify
+the JSON in the format of:
+
+```json
+servicMethodName: {
+  "service": "<WSDL service name>",
+  "port": "<WSDL port name>",
+  "operation": "<WSDL operation name>"
+}
+```
+
+If you have more than one operations to map, it might be easier to edit the
+DataSource JSON after it's been created. See below for the example of the
+mapping of the WSDL binding operations and Node.js methods.
+
+```json
+{
+  "name": "ds",
+  "connector": "soap",
+  "url": "http://calculator-webservice.mybluemix.net/calculator",
+  "wsdl": "http://calculator-webservice.mybluemix.net/calculator?wsdl",
+  "remotingEnabled": true,
+  // ADD THIS SNIPPET
+  "operations": {
+    "add": {
+      "service": "CalculatorService",
+      "port": "CalculatorPort",
+      "operation": "add"
+    },
+    "subtract": {
+      "service": "CalculatorService",
+      "port": "CalculatorPort",
+      "operation": "subtract"
+    }
+  }
+  // END OF THE SNIPPET
+}
+```
+
+For details, you can refer to the SOAP connector's operations property:
+https://github.com/strongloop/loopback-connector-soap#operations-property
```

Comment:
will change it to https://loopback.io/doc/en/lb3/SOAP-connector.html#operations-property.